ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Mock/Block.hs

```diff
@@ -220,21 +220,29 @@ instance (SimpleCrypto c, Typeable ext, SupportedBlock (SimpleBlock c ext))
 
   type LedgerError (SimpleBlock c ext) = MockError (SimpleBlock c ext)
 
-  applyLedgerHeader _cfg hdr (SimpleLedgerState st) =
-      SimpleLedgerState <$> updateMockTip hdr st
-  applyLedgerBlock  _cfg = updateSimpleLedgerState
+  applyChainTick _ _ = return
+  applyLedgerBlock _cfg = updateSimpleLedgerState
   ledgerTipPoint (SimpleLedgerState st) = mockTip st
   ledgerConfigView _ = SimpleLedgerConfig
 
-updateSimpleLedgerState :: (Monad m, HasUtxo a)
-                        => a
+updateSimpleLedgerState :: (Monad m, SimpleCrypto c, Typeable ext)
```

Comment:
How do these changes fit in with the rest of the PR?

Reply:
`updateSimpleLedgerState` needed to combine header and body processing, as it was only doing body before. The change had further reaching consequences than expected, but the `updateSimpleUTxO` helped to keep it simple.

Reply:
Gotcha :-)